lib/std/os/windows.zig

```diff
@@ -3789,6 +3789,8 @@ pub const PATH_MAX_WIDE = 32767;
 /// TODO: More verification of this assumption.
 pub const NAME_MAX = 255;
 
+pub const HOST_NAME_MAX = 256;
```

Comment:
```suggestion
/// > The maximum length, in bytes, of the string returned in the buffer
/// > pointed to by the name parameter is dependent on the namespace
/// > provider, but this string must be 256 bytes or less. So if a buffer of
/// > 256 bytes is passed in the name parameter and the namelen parameter
/// > is set to 256, the buffer size will always be adequate.
/// from https://learn.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-gethostname
pub const HOST_NAME_MAX = 256;
```

It also seems like the 256 bytes must include the `NUL` terminator. Would be nice to know that for sure and include that info in the doc comment as well.

Reply:
I am under the belief that 256 does include it nul-terminator. reading the comment again leads me to believe this Is the code they were envisioning when writing it:

```c
#include <unistd.h>

int main(int argc, char* argv[]) {
    unsigned char name[256];
    gethostname(name, 256);
    return 0;
}
```

additionally, https://man7.org/linux/man-pages/man2/gethostname.2.html has a remark that:

> SUSv2 guarantees that "Host names are limited to 255 bytes". POSIX.1 guarantees that "Host names (not including the terminating null byte) are limited to HOST_NAME_MAX bytes".

Reply:
Agreed that it's very likely that the 256 includes the terminator. I think setting the constant to 255 makes the most sense here, since most/all other HOST_NAME_MAX definitions are 255 (i.e. don't include the terminator).

(relatedly, but can be addressed in a separate PR, [`std.os.gethostname`](https://github.com/ziglang/zig/blob/63de8a59891f2c342d1a51b808cf04a895ba54d6/lib/std/os.zig#L5788-L5807) is currently incorrectly handling the buffer size with regards to the `NUL` terminator since its buffer does not include room for HOST_NAME_MAX bytes + a `NUL` terminator; it should probably take a buffer of `[HOST_NAME_MAX:0]u8` and return `[:0]u8`)